CRM/Core/Resources.php

```diff
@@ -881,4 +882,27 @@ private function resolveFileName(&$fileName, $extName) {
     }
   }
 
+  /**
+   * @param string $url
+   * @return string
+   */
+  public function addCacheCode($url) {
+    parse_str(parse_url($url, PHP_URL_QUERY), $queryParts);
+    $existing = isset($queryParts['r']) ? $queryParts['r'] : NULL;
+    $latest = $this->cacheCode;
+
+    if ($existing) {
+      if ($existing === $latest) {
+        return $url; // no need to update
+      }
+      else {
+        return str_replace('r=' . $existing, 'r=' . $latest, $url);
```

Comment:
Yeah, that rewrite/simplification sounds good.

The cache busting string does, of course, change from time-to-time, but I look at it this way:
 * The signature for `Resources::getUrl()` says that it takes an extension-name and file-name (not a URL fragment). The intended usage is this: someone calls `addScriptFile(...)` or `addStyleFile(...)`, then it automatically composes the URL and outputs the markup. There's no opportunity in that flow for someone to append their own query-parameters or re-append the cache-code.
 * As a general matter, the code is and should be stable because we *want* caching to work. The cache code should only change in a few limited cases: (a) the sysadmin does some kind of system update (upgrade core; install a new extension) or (b) the sysadmin/developer explicitly flushes the caches.